src/main/java/data/plans/PlanList.java

```diff
@@ -174,4 +176,36 @@ public void listAllPlan() {
             System.out.println((i + 1) + "." + getPlansDisplayList().get(i).toString());
         }
     }
+
+    /**
+     * Removes the intended plan in the plans list.
+     * The plan to delete is determined by the user who will
+     * indicate the index of plan to delete in the plan list.
+     *
+     * @param userArgument The argument entered by user, that is, the index of plan to delete.
+     * @return deletedPlan The plan object that is deleted from the plansDisplayList.
+     * @throws NumberFormatException If index of plan that user entered is not an integer.
+     * @throws InvalidPlanException If index of plan to delete is out of range.
+     */
+    public Plan deletePlan(String userArgument) throws NumberFormatException, InvalidPlanException {
+        int indexToDelete = Integer.parseInt(userArgument.trim());
+        String className = this.getClass().getSimpleName();
+
+        boolean isIndexToDeleteValid = checkIndexIsWithinRange(indexToDelete);
+
+        if (!isIndexToDeleteValid) {
+            throw new InvalidPlanException(className, InvalidPlanException.PLAN_INDEX_OUT_OF_RANGE);
+        }
+
+        assert (indexToDelete > 0) && (indexToDelete <= plansDisplayList.size());
+        Plan deletedPlan = getPlanFromIndexNum(indexToDelete);
+        String deletedPlanKey = deletedPlan.toString();
+        plansDisplayList.remove(indexToDelete - 1);
+        getPlansDisplayList().remove(deletedPlanKey);
```

Comment:
is this supposed to be getPlansHashMapList() as I notice only the plan gets removed from the plans list (display list), not the hashmap?

Reply:
Yes, it should be getPlansHashMapList(). I didn't notice that. Thank you for pointing it out.